multiversx_sdk/network_providers/api_network_provider.py

```diff
@@ -125,10 +124,21 @@ def get_non_fungible_token(self, collection: str, nonce: int) -> NonFungibleToke
         result = NonFungibleTokenOfAccountOnNetwork.from_api_http_response(response)
         return result
 
-    def query_contract(self, query: IContractQuery) -> ContractQueryResponse:
-        request = ContractQueryRequest(query).to_http_request()
+    def query_contract(self, query: SmartContractQuery) -> SmartContractQueryResponse:
+        query_converter = SmartContractQueryConverter()
+        request = query_converter.smart_contract_query_to_dictionary(query)
         response = self.do_post_generic('query', request)
-        return ContractQueryResponse.from_http_response(response)
+
+        return_data = response.get('returnData', []) or response.get('ReturnData', [])
+        return_code = response.get('returnCode', '') or response.get('ReturnCode', '')
+        return_message = response.get('returnMessage', '') or response.get('ReturnMessage', '')
```

Comment:
Somehow duplication, but can be kept as it is for now (until we tackle the design asymmetry mentioned in the other comment).

Reply:
As discusses, transformations and conversions can be done in `shared.py` or something similar (not in the converters package).